ion/symboltoken.go

```diff
@@ -97,14 +100,38 @@ func (st *SymbolToken) Equal(o *SymbolToken) bool {
 	return false
 }
 
+func getSystemSymbolMapping(symbolTable SymbolTable, symbolName string) string {
+	// If we have a symbol name of the form '$n' for some integer n,
+	// then we want to use the corresponding system symbol name.
+	if len(symbolName) > 1 && symbolName[0] == '$' {
+		if id, err := strconv.Atoi(symbolName[1:]); err == nil {
+			if systemSymbolName, ok := symbolTable.FindByID(uint64(id)); ok {
+				return systemSymbolName
+			}
+		}
+	}
+
+	return ""
+}
+
 // NewSymbolToken will check and return a symbol token if it exists in a symbol table,
 // otherwise return a new symbol token.
 func NewSymbolToken(symbolTable SymbolTable, text string) SymbolToken {
+	if symbolTable == nil {
+		symbolTable = V1SystemSymbolTable
+	}
+
+	systemSymbolName := getSystemSymbolMapping(symbolTable, text)
+	if systemSymbolName != "" {
+		return SymbolToken{Text: &systemSymbolName, LocalSID: SymbolIDUnknown}
```

Comment:
If the system symbol table contains the text in question, why is the `LocalSID` unknown?

Reply:
Addressed.